tests/test_steps.py

```diff
@@ -70,12 +70,18 @@ def temporary_step() -> Iterator[str]:
             ipy_file.unlink()
 
 
-def test_topological_sort():
+def test_dependency_ordering():
     "Check that a dependency will be scheduled to run before things that need it."
     dag = {"a": ["b", "c"], "b": ["c"]}
     assert to_dependency_order(dag, [], []) == ["c", "b", "a"]
 
 
+def test_dependency_filtering():
+    dag = {"e": {"a"}, "c": {"b", "d"}, "b": {"a"}}
```

Comment:
Same DAG as @larsyencken 's diagrams (super helpful by the way!) - just opposite direction convention (see comment above on reversing)